src/context/data-context.js

```diff
@@ -0,0 +1,57 @@
+import { createContext, useContext, useReducer, useEffect } from "react";
+import axios from "axios";
+import { reducer } from "./data-reducer";
+
+export const DataContext = createContext();
+
+export function DataProvider({ children }) {
+  const [data, dispatch] = useReducer(reducer, {
+    cartList: [],
+    wishList: [],
+    productList: [],
+    listManipulation: {
+      sortBy: null,
+      fastDelivery: false,
+      outOfStock: true,
+      bestSeller: false
+    },
+    navToggle: false,
+    navSidebarToggle: false
+  });
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const { data } = await axios.get("/api/products");
+        dispatch({ type: "LOAD_PRODUCT_LIST", payload: data.products });
+      } catch (error) {console.log(error)}
+    })();
+  }, []);
+
+  const getSortedData = (productList, sortby) => {
+    if (sortby === "SORT_PRICE_ASC")
+      return productList.sort((a, b) => a.price - b.price);
+    else if (sortby === "SORT_PRICE_DESC")
+      return productList.sort((a, b) => b.price - a.price);
```

Comment:
this will mutate the original productList array. To check its effect, u can try doing a Clear Filter btn. Hence we should apply `sort` on a new array. Here's a way we can do it:
`[...productList].sort((a, b) => a.price - b.price);`